pkg/swagger/swagger.go

```diff
@@ -50,19 +55,31 @@ func Run(api, outputDir string) error {
 			Summary:     "Lists credentials of an OpenShift cluster with the specified subscription, resource group and resource name.",
 			Description: "The operation returns the credentials.",
 			OperationID: "OpenShiftClusters_ListCredentials",
-			Parameters:  populateParameters(3, "OpenShiftCluster", "OpenShift cluster"),
-			Responses:   populateResponses("OpenShiftClusterCredentials", false, http.StatusOK),
+			Parameters:  g.populateParameters(3, "OpenShiftCluster", "OpenShift cluster"),
+			Responses:   g.populateResponses("OpenShiftClusterCredentials", false, http.StatusOK),
 		},
 	}
+	if g.kubeconfig {
+		s.Paths["/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RedHatOpenShift/openShiftClusters/{resourceName}/listAdminCredentials"] = &PathItem{
```

Comment:
Yes, this will be different endpoint and API.  Few reasons for this:
1. Prevent api breakage for old API
2. API purpose is different as it will download file, and not expose
3. Easier to implement. 
4. It will return privileged kubeconfig so in the future dedicated rbac could be added